lib/slack/real_time/concurrency/async.rb

```diff
@@ -6,40 +6,45 @@ module Concurrency
       module Async
         class Client < ::Async::WebSocket::Client
           extend ::Forwardable
-          def_delegators :driver, :on
+          def_delegators :@driver, :on
 
           def text(message)
-            driver.text(message)
+            @driver.text(message)
           end
 
           def binary(data)
-            socket.write(data)
+            @driver.binary(data)
           end
         end
 
         class Socket < Slack::RealTime::Socket
           attr_reader :client
 
           def start_async(client)
-            @client = client
-            client.run_loop
+            Thread.new do
```

Comment:
We generally want the caller to be responsible for calling this, otherwise we're spawing a thread per client (or a bot).